### PR TITLE
swipe up/down history

### PR DESCRIPTION
**Pull request by chescalante**

### CODE REVIEW

File: apps/daimo-mobile/package.json [hunk 1]
```diff
@@ -66,6 +66,7 @@
     "react-native-safe-area-context": "4.6.3",
     "react-native-screens": "~3.22.0",
     "react-native-svg": "13.9.0",
+    "react-native-swipe-up-down": "github:react-native-vietnam/react-native-swipe-up-down",
```
Comment (dcposch) [inline apps/daimo-mobile/package.json:69]: is this different from the version on NPM?

Comment (dcposch) [reply apps/daimo-mobile/package.json:69]: pros: looks like a simple component https://www.npmjs.com/package/react-native-swipe-up-down?activeTab=code

cons: ~200 downloads a month, almost nobody is using this

Comment (dcposch) [reply apps/daimo-mobile/package.json:69]: have you tried https://www.npmjs.com/package/@gorhom/bottom-sheet

File: apps/daimo-mobile/src/view/screen/HomeScreen.tsx [hunk 2]
```diff
@@ -26,43 +21,45 @@ export default function HomeScreen() {
...
-    <View style={ss.container.outerStretchScroll}>
-      <Header />
-
-      <AmountAndButtons account={account} />
+    <View style={ss.container.homeContainer}>
+      <View
+        style={{
+          paddingLeft: 50,
+          paddingRight: 50,
+        }}
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/HomeScreen.tsx:32]: recommend pulling this out into a `StyleSheet`, see other components

File: apps/daimo-mobile/src/view/screen/HomeScreen.tsx [hunk 3]
```diff
@@ -26,43 +21,45 @@ export default function HomeScreen() {
...
+        itemFull={<HistoryScreen />}
+        onShowMini={() => console.log("mini")}
+        onShowFull={() => console.log("full")}
+        animation="spring"
+        disableSwipeIcon
+        extraMarginTop={0}
+        iconColor="blue"
+        iconSize={30}
+        swipeHeight={300}
+        style={{ backgroundColor: "#fff" }}
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/HomeScreen.tsx:61]: can also go in StyleSheet

File: apps/daimo-mobile/src/view/screen/HomeScreen.tsx [hunk 4]
```diff
@@ -26,43 +21,45 @@ export default function HomeScreen() {
...
+      <SwipeUpDown
+        itemMini={
+          <View style={styles.historyElem}>
+            <ScrollPellet />
+            <HistoryList account={account} maxToShow={5} />
+          </View>
+        }
+        itemFull={<HistoryScreen />}
+        onShowMini={() => console.log("mini")}
+        onShowFull={() => console.log("full")}
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/HomeScreen.tsx:54]: delete logs

Comment (chescalante) [reply apps/daimo-mobile/src/view/screen/HomeScreen.tsx]: done

File: apps/daimo-mobile/src/view/screen/HomeScreen.tsx (hunk 3 above)
Comment (chescalante) [reply apps/daimo-mobile/src/view/screen/HomeScreen.tsx]: done

File: apps/daimo-mobile/src/view/screen/HomeScreen.tsx (hunk 2 above)
Comment (chescalante) [reply apps/daimo-mobile/src/view/screen/HomeScreen.tsx]: done

File: apps/daimo-mobile/package.json (hunk 1 above)
Comment (chescalante) [reply apps/daimo-mobile/package.json]: i was trying a couple of thing but now is installed from npm normally. We can discuss about using another package if you want.

File: apps/daimo-mobile/src/view/screen/HomeScreen.tsx [hunk 5]
```diff
@@ -32,44 +27,46 @@ export default function HomeScreen() {
...
-        <View style={styles.historyElem}>
-          <ScrollPellet />
-          <HistoryList account={account} maxToShow={5} />
-        </View>
-      </ScrollView>
+    <View
+      style={{
+        ...ss.container.homeContainer,
+        top: -addTopOffset,
+        paddingTop: addTopOffset,
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/HomeScreen.tsx:44]: this part is a bit rough, but fine if no other way

File: apps/daimo-mobile/src/view/screen/HomeScreen.tsx [hunk 6]
```diff
@@ -32,44 +27,46 @@ export default function HomeScreen() {
...
+            <HistoryList account={account} maxToShow={5} />
+          </View>
+        }
+        itemFull={<HistoryScreen />}
+        onShowMini={() => setIsHistoryOpened(false)}
+        onShowFull={() => setIsHistoryOpened(true)}
+        animation="spring"
+        disableSwipeIcon
+        extraMarginTop={0}
+        swipeHeight={300}
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/HomeScreen.tsx:67]: will adjust this height and the stylesheet below

**Review by dcposch — COMMENTED** [review]
Rebased, squashed. Overall, looks great to me--thanks again.

**Comment by dcposch** [comment]
https://github.com/daimo-eth/daimo/assets/169280/ec6fd874-c159-4c8a-8a07-78676509ffba

**Pull request merged**